src/components/v5/common/ActionSidebar/partials/ExpenditureBadge/hooks.ts

```diff
@@ -0,0 +1,73 @@
+import { useMemo } from 'react';
+
+import { ExpenditureStatus } from '~gql';
+import { type Expenditure } from '~types/graphql.ts';
+import { isExpenditureFullyFunded } from '~v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts';
+
+import { ExpenditureActionStatus } from './types.ts';
+
+export const useExpenditureActionStatus = (
+  expenditure: Expenditure | null | undefined,
+  withAdditionalStatuses?: boolean,
+): ExpenditureActionStatus => {
+  // @todo: update when edit mode will be implemented
+  const paymentIsInEditMode = false;
+  const paymentHasAnActiveEditRequest = false;
+
+  // @todo: update when cancel with reputation decision method will be implemented
+  const paymentHasAnActiveCancelRequest = false;
+
+  const { status, slots } = expenditure || {};
+
+  const allPayouts = useMemo(
+    () =>
+      slots?.flatMap(
+        ({ payouts, claimDelay, id }) =>
+          payouts?.map((payout) => ({
+            ...payout,
+            slotId: id,
+            claimDelay: claimDelay || '0',
```

Comment:
Do we need any of those properties? Looks like the code below is only checking `isClaimed` property